run_dir/static/js/project_samples.js

```diff
@@ -368,20 +398,29 @@ function load_all_udfs(){
           var pi = data['order_details']['fields']['project_pi_email'];
           if(!emails[pi]){emails[pi]=[elabel('PI', 'default')]} else {emails[pi].push(elabel('PI', 'default'))};
 
-          email_html = '';
           Object.keys(emails).forEach(function(k, i) {
-            if (k != 'null'){ // A bit ugly ¯\_(ツ)_/¯
+            if (!validateEmail(k)) {
+              console.log('Found a strange email address, falling back to old method, '+k);
+              throw 'TypeError';
+            }
+          });
+          Object.keys(emails).forEach(function(k, i) {
```

Comment:
Why not continue in the previous loop ? Can we just set the `email_html = ''` before throwing error if your concern is that it might fail in the middle way and have incomplete data ?

Reply:
That's exacly my concern. This only adds 4 iterations and 6 lines of code. I couldn't be bothered to do anything more clever 😀